database_admin/migrations/025_packages.up.sql

```diff
@@ -0,0 +1,54 @@
+CREATE TABLE IF NOT EXISTS package_name
+(
+    id   INT GENERATED BY DEFAULT AS IDENTITY NOT NULL PRIMARY KEY,
+    name TEXT                                 NOT NULL CHECK ( NOT empty(name)) UNIQUE
+);
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE package_name TO vmaas_sync;
+GRANT SELECT ON TABLE package_name TO evaluator;
+GRANT SELECT ON TABLE package_name TO listener;
+GRANT SELECT ON TABLE package_name TO manager;
+
+GRANT SELECT, USAGE ON SEQUENCE package_name_id_seq TO evaluator;
+GRANT SELECT, USAGE ON SEQUENCE package_name_id_seq TO listener;
+GRANT SELECT, USAGE ON SEQUENCE package_name_id_seq TO vmaas_sync;
+
+CREATE TABLE IF NOT EXISTS package
+(
+    id          INT GENERATED BY DEFAULT AS IDENTITY NOT NULL PRIMARY KEY,
+    -- TODO: inline vs extract ?
+    name_id     INT                                  NOT NULL REFERENCES package_name,
+    version     TEXT                                 NOT NULL CHECK (NOT empty(version)),
+    -- TODO: Inline vs extract to table ?
+    description TEXT                                 NOT NULL CHECK ( NOT empty(description)),
+    -- TODO: Inline vs extract to table ?
+    summary     TEXT                                 NOT NULL CHECK ( NOT empty(summary)),
+    UNIQUE (name_id, version)
+);
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE package TO vmaas_sync;
+GRANT SELECT ON TABLE package TO evaluator;
+GRANT SELECT ON TABLE package TO listener;
+GRANT SELECT ON TABLE package TO manager;
+
+GRANT SELECT, USAGE ON SEQUENCE package_id_seq TO evaluator;
+GRANT SELECT, USAGE ON SEQUENCE package_id_seq TO listener;
+GRANT SELECT, USAGE ON SEQUENCE package_id_seq TO vmaas_sync;
+
+CREATE TABLE IF NOT EXISTS system_package
+(
+    system_id   INT NOT NULL REFERENCES system_platform,
+    package_id  INT NOT NULL REFERENCES package,
+    -- Use null to represent up-to-date packages
+    update_data JSONB DEFAULT NULL,
```

Comment:
What do you want to store in this "update_data" JSONB? I think we don't need it. We store all update informations by pairing system-package in this table. Maybe we can add "current_version" column to store info about currently installed package version. And so we can add "package_version" enumerating table for that (also used in "package" table). Because this system_package pairing table will be probably huge.

Reply:
Because if we store updates in separate table, the primary key for that table will be `system_id, package_id, index`. I thing for now it's better to do it this way, because of the size of that table, and issues when querying for data.

Reply:
Why? As I know we need to store only the latest version for each package because only last update is required.

Reply:
Yes, that is the current state, but in the future we might want to show all updates.

Reply:
ok, it's all update version.